cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/update/StateVariablesAdder.java

```diff
@@ -100,20 +210,109 @@ public static void add(Network n, CgmesModel cgmes) {
+                if (leg.getPhaseTapChanger() != null) {
+                    p.put(tapChangerPositionName, is(leg.getPhaseTapChanger().getTapPosition()));
+                    p.put(CgmesNames.TAP_CHANGER, cgmes.phaseTapChangerForPowerTransformer(t.getId()));
+                    tapSteps.add(p);
+                } else if (leg.getRatioTapChanger() != null) {
+                    p.put(tapChangerPositionName, is(leg.getRatioTapChanger().getTapPosition()));
+                    p.put(CgmesNames.TAP_CHANGER, cgmes.ratioTapChangerForPowerTransformer(t.getId()));
+                    tapSteps.add(p);
+                }
+            });
+        }
+
+        cgmes.add(originalSVcontext, "SvTapStep", tapSteps);
+    }
+
+    private void addStatusToCgmes() {
+        // create SvStatus, iterate on Connectables, check Terminal status, add
+        // to SvStatus
+        PropertyBags svStatus = new PropertyBags();
+        Set<String> addedConnectables = new HashSet<>();
+        for (VoltageLevel v : n.getVoltageLevels()) {
+            for (Connectable<?> c : v.getConnectables()) {
+                // need to check if connectable was already added
+                if (!addedConnectables.contains(c.getId())) {
+                    addedConnectables.add(c.getId());
+                    if (c.getId() != null) {
+                        PropertyBag p = new PropertyBag(SV_SVSTATUS_PROPERTIES);
+                        for (Terminal t : c.getTerminals()) {
+                            if (t == null) {
+                                continue;
+                            }
+                            p.put(IN_SERVICE, String.valueOf(t.isConnected()));
+                        }
+                        p.put(CgmesNames.CONDUCTING_EQUIPMENT, c.getId());
+                        svStatus.add(p);
+                    }
+                }
+            }
+        }
+
+        // SvStatus at boundaries set as it was in original cgmes.
+        for (PropertyBag terminal : originalTerminals) {
+            Objects.requireNonNull(terminal);
+            if (terminal.getId("SvStatus") == null) {
+                continue;
+            }
+            boundaryNodesFromDanglingLines.values().forEach(value -> {
+                if (terminal.getId(CgmesNames.TOPOLOGICAL_NODE).equals(value)) {
+                    PropertyBag p = new PropertyBag(SV_SVSTATUS_PROPERTIES);
+                    p.put(IN_SERVICE, terminal.getId(IN_SERVICE));
+                    p.put(CgmesNames.CONDUCTING_EQUIPMENT, terminal.getId(CgmesNames.CONDUCTING_EQUIPMENT));
+                    svStatus.add(p);
+                }
+            });
+        }
+        cgmes.add(originalSVcontext, "SvStatus", svStatus);
+    }
+
+    private Map<String, String> boundaryNodesFromDanglingLines() {
+        Map<String, String> nodesFromLines = new HashMap<>();
+        List<String> boundaryNodes = new ArrayList<>();
+        cgmes.boundaryNodes().forEach(node -> boundaryNodes.add(node.getId("Node")));
```

Comment:
Use the stream API to collect boundadyNodes:
```
List<String> boundaryNodes = cgmes.boundaryNodes().map(X::getId).collect(Collectors.toList());
```
Replace X by the name of the class of `node`.

Reply:
Done. With the only difference: we we used arrow operator, not double colon
List<String> boundaryNodes = cgmes.boundaryNodes().stream().map(n -> n.getId("Node")).collect(Collectors.toList());